src/LaunchDarkly.EventSource/EventSourceService.cs

```diff
@@ -228,46 +228,23 @@ private HttpRequestMessage CreateHttpRequestMessage(Uri uri, string lastEventId)
             return request;
         }
 
-        private void HandleInvalidResponses(HttpResponseMessage response)
+        private void ValidateResponse(HttpResponseMessage response)
         {
-            HandleUnsuccessfulStatusCodes(response);
-
-            // According to Specs, a client can be told to stop reconnecting using the HTTP 204 No Content response code
-            HandleNoContent(response);
-
-            // According to Specs, HTTP 200 OK responses that have a Content-Type specifying an unsupported type, 
-            // or that have no Content-Type at all, must cause the user agent to fail the connection.
-            HandleIncorrectMediaType(response);
-        }
-
-        private void HandleNoContent(HttpResponseMessage response)
-        {
-            if (response.StatusCode == System.Net.HttpStatusCode.NoContent)
+            // Any non-2xx response status is an error. A 204 (no content) is also an error.
+            if (!response.IsSuccessStatusCode || response.StatusCode == System.Net.HttpStatusCode.NoContent)
             {
-                throw new EventSourceServiceUnsuccessfulResponseException(Resources.EventSource_204_Response, (int)response.StatusCode);
+                throw new EventSourceServiceUnsuccessfulResponseException((int)response.StatusCode);
```

Comment:
There's no longer a separate error message for complaining about a 204 response; it's just another unexpected status code. There can be a special meaning for 204 in SSE-based applications (the convention is that it means "don't bother trying to reconnect, there won't ever be any content here") but it's up to the caller to make decisions about that, which they can already do by looking at the StatusCode property of the exception.